virtualmachine/gcp/vm.go

```diff
@@ -56,17 +65,127 @@ type VM struct {
 
 	AccountFile  string
 	account      accountFile
-	SSHCreds     ssh.Credentials // privateKey is required for GCE
+	SSHCreds     ssh.Credentials
 	SSHPublicKey string
+
+	Firewall  string // required when modifying firewall rules
+	Endpoints []Endpoint
+}
+
+// Endpoint represents the protocol and ports configured in a firewall
+type Endpoint struct {
+	// Protocol can be one of the following well known protocol strings
+	// (tcp, udp, icmp, esp, ah, sctp) which are supported by GCP
+	Protocol string
+	Ports    []string
+}
+
+// Image represents a GCE image
+type Image struct {
+	Id                uint64 `json:"id,omitempty,string"`
+	CreationTimestamp string `json:"creation_timestamp,omitempty"`
+	// DeprecationStatus indicates whether the image is depcrecated. If the
+	// image is not deprecated it will have an empty string.
+	DeprecationStatus string `json:"deprecation_status,omitempty"`
+	// DeprecationReplacement provides the url of the image which is a
+	// replacement of this deprecated image. If this image is not deprecated
+	// it's value will be an empty string.
+	DeprecationReplacement string `json:"deprecation_replacement,omitempty"`
+	Description            string `json:"description,omitempty"`
+	DiskSizeGb             int64  `json:"disk_size_gb,omitempty,string"`
+	Family                 string `json:"family,omitempty"`
+	Name                   string `json:"name,omitempty"`
+	Status                 string `json:"status,omitempty"`
 }
 
 // Disk represents the GCP Disk.
 // See https://cloud.google.com/compute/docs/disks/?hl=en_US&_ga=1.115106433.702756738.1463769954
 type Disk struct {
-	Name       string
-	DiskType   string
-	DiskSizeGb int
-	AutoDelete bool // Auto delete disk
+	Name        string
+	DiskType    string
+	DiskSizeGb  int
+	AutoDelete  bool // Auto delete disk
+	Description string
+}
+
+// AttachedDisk represents a disk attached to a GCE instance
+type AttachedDisk struct {
+	Name       string `json:"name,omitempty"`
+	DevicePath string `json:"device_path,omitempty"`
+	Boot       *bool  `json:"boot,omitempty"`
+	AutoDelete *bool  `json:"auto_delete,omitempty"`
+	Interface  string `json:"interface,omitempty"`
```

Comment:
Does it map to the instance?
Please add one-liner description.

Reply:
From GCP documentation: Specifies the disk interface, which is either SCSI or NVME. The default is SCSI. Persistent disks must always use SCSI and the request will fail if you attempt to attach a persistent disk in any other format than SCSI. Local SSDs can use either NVME or SCSI.
Adding comment to describe.